src/main/java/net/glowstone/util/PathFinder.java

```diff
@@ -0,0 +1,243 @@
+package net.glowstone.util;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.PriorityQueue;
+import java.util.Queue;
+import java.util.Set;
+
+import org.bukkit.util.Vector;
+
+/**
+ * General class for path finding.
+ * TODO: Implement other path finding algorithms (BFS, DFS, D*, potentially more)
+ */
+
+public class PathFinder {
+
+    private final Vector startNode;
+    private final Vector endNode;
+    private List<Vector> path;
+    private Map<Vector, Vector> fromPath;
```

Comment:
Can we map the vectors into longs so that we can be slightly more efficient (with trove)?